src/modules/0.0.16/aws_account/index.ts

```diff
@@ -1,50 +1,49 @@
 import { AWS, } from '../../../services/aws_macros'
 import { AwsAccountEntity, } from './entity'
-import { Context, Crud2, Mapper2, Module2, } from '../../interfaces'
-import * as metadata from './module.json'
+import { Context, Crud2, Mapper2, ModuleBase, } from '../../interfaces'
+import * as metadata from './module.json' // TODO: Eliminate this?
 
-export const AwsAccount: Module2 = new Module2({
-  ...metadata,
-  provides: {
-    context: {
-      // This function is `async function () {` instead of `async () => {` because that enables the
-      // `this` keyword within the function based on the objec it is being called from, so the
-      // `getAwsClient` function can access the correct `orm` object with the appropriate creds and
-      // read out the right AWS creds and create an AWS client also attached to the current context,
-      // which will be different for different users. WARNING: Explicitly trying to access via
-      // `AwsAccount.provides.context.getAwsClient` would instead use the context *template* that is
-      // global to the codebase.
-      async getAwsClient() {
-        const orm = this.orm;
-        if (this.awsClient) return this.awsClient;
-        const awsCreds = await orm.findOne(AwsAccount.mappers.awsAccount.entity);
-        this.awsClient = new AWS({
-          region: awsCreds.region,
-          credentials: {
-            accessKeyId: awsCreds.accessKeyId,
-            secretAccessKey: awsCreds.secretAccessKey,
-          },
-        });
-        return this.awsClient;
-      },
-      awsClient: null, // Just reserving this name to guard against collisions between modules.
+class AwsAccount extends ModuleBase {
+  constructor() { super(); super.init(); }
+  dirname = __dirname;
```

Comment:
do you foresee a case where `dirname` is not `__dirname`, might be weird to add this in all module's definitions?

Reply:
It is already in all modules, just at the bottom instead of the top (see the end of the diff on this file). Unfortunately, because `__dirname` is a magic variable injected by Node.js itself, I can't hide this from the developers. If I use `__dirname` in the `ModuleBase` base class, it'll always be `src/modules/` because it's inside of the `src/modules/interfaces.ts` file.